web/src/main/java/ua/com/vertex/controllers/CertificateDetailsPageController.java

```diff
@@ -29,50 +28,31 @@ public String showCertificateDetailsPage() {
     }
 
     @RequestMapping(value = "/getCertificate")
-    public String getCertificate(@RequestParam String certificateIdEncoded, Model model) {
-
-        String view = CERTIFICATE_DETAILS;
-
-        int certificateId = certLogic.decodeId(certificateIdEncoded, model);
-        if (certificateId == WRONG_ID) {
-            model.addAttribute(ERROR, "Invalid entered data");
-            LOGGER.debug(logInfo.getId() + "Entered certificateId data are invalid");
-        } else {
-            try {
-                certLogic.setUserAndCertificate(certificateId, model);
-                LOGGER.debug(logInfo.getId() + "Passing certificate and user data to JSP");
-            } catch (Exception e) {
-                LOGGER.warn(logInfo.getId(), e, e);
-                view = ERROR;
-            }
-        }
-
-        return view;
+    public String getCertificate(@RequestParam String certificateUid, Model model) {
+        return process(certificateUid, model);
     }
 
-    @RequestMapping(value = "/getCertificate/{certificateIdEncoded}")
-    public String getCertificateByCertificateId(@PathVariable String certificateIdEncoded, Model model) {
+    @RequestMapping(value = "/getCertificate/{certificateUid}")
+    public String getCertificateByCertificateUid(@PathVariable String certificateUid, Model model) {
+        return process(certificateUid, model);
+    }
 
+    private String process(String certificateUid, Model model) {
         String view = CERTIFICATE_DETAILS;
 
-        int certificateId = certLogic.decodeId(certificateIdEncoded, model);
-        if (certificateId == WRONG_ID) {
-            model.addAttribute(ERROR, "Invalid entered data");
-            LOGGER.debug(logInfo.getId() + "Entered certificateId data are invalid");
-        } else {
-            try {
-                certLogic.setUserAndCertificate(certificateId, model);
-            } catch (Exception e) {
-                LOGGER.warn(logInfo.getId(), e, e);
-                view = ERROR;
-            }
+        certificateUid = certificateUid.replaceAll("-", "");
+        try {
+            certLogic.getUserAndCertificate(certificateUid, model);
```

Comment:
If the problem is transferring model to service layer and filling it there, then I can provide this variant: http://clip2net.com/s/3KKSb67